input/kafkamdm/kafkamdm.go

```diff
@@ -218,6 +218,27 @@ func (k *KafkaMdm) Start(handler input.Handler) {
 	}
 }
 
+// tryGetOffset will query kafka repeatedly for the requested offset, and return it when it has it,
+// or halt the program if can't retrieve the value
+func (k *KafkaMdm) tryGetOffset(topic string, partition int32, offsetStr string, offset int64, attempts int, sleep time.Duration) int64 {
```

Comment:
offsetStr in something that should be determined inside this function, not passed in.

```
var offsetStr string
switch offset {
case sarama.OffsetNewest:
  offsetStr = "newest"
case sarama.OffsetOldest:
  offsetStr = "oldest"
default:
  offsetStr = strings.FormatInt(offset, 10)
}